### PR TITLE
chore(weave): add width to autocomplete

### PR DESCRIPTION
**Pull request by mbarrramsey**
## Description

- Fixes WB-NNNNN
- Fixes #NNNN

Adds minWidth param and ref to the autocomplete component so we can measure the full width of the text input for use in https://github.com/wandb/core/pull/24247.

## Testing

How was this PR tested?

### CODE REVIEW

**Review by connieelee — APPROVED** [review]

File: weave-js/src/components/Form/AutoComplete.tsx [hunk 1]
```diff
@@ -145,15 +145,17 @@ type AdditionalProps = {
   size?: SelectSize;
   isDarkMode?: boolean;
   maxWidth?: number;
+  minWidth?: number;
   hasInputValue?: boolean;
+  inputRef?: React.MutableRefObject<HTMLDivElement | null>;
```
Comment (connieelee) [inline weave-js/src/components/Form/AutoComplete.tsx:150]: **nitpick:** `inputRef` naming seems misleading if it's actually a div element under there?

Comment (mbarrramsey) [reply weave-js/src/components/Form/AutoComplete.tsx:150]: Maybe `inputDivRef`? or `autocompleteDivRef`?

Comment (connieelee) [reply weave-js/src/components/Form/AutoComplete.tsx:150]: we should probably leave out "input" entirely. if you inspect the elements, there's an actual `input` element nested inside, but this ref is definitely not being attached to that.

have you tried `React.forwardRef()`?

Comment (mbarrramsey) [reply weave-js/src/components/Form/AutoComplete.tsx:150]: I did try the forwardRef, but I don't think it works well with generics 🤔 see https://github.com/wandb/core/pull/24247/files#r1777844929. We could try using the wrapper approach but it seems like the custom name is similar?

Comment (connieelee) [reply weave-js/src/components/Form/AutoComplete.tsx:150]: ahh i see. maybe something like `containerRef` or `autocompleteRef` seems fine then!

**Pull request merged**